### PR TITLE
feat(Widget): Create foundations for the graph widget

### PR DESCRIPTION
**Pull request by Thebarda**
## Description

This implements the metrics data selection system for the graph widget.

https://github.com/centreon/centreon/assets/12515407/bae2091d-bd9a-4f5e-a9e3-51e865220433

https://github.com/centreon/centreon/assets/12515407/c1d6d2f9-18bd-4cd4-b8fa-87ca2adec5dc

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by bdauria — DISMISSED** [review]

**Review by bdauria — DISMISSED** [review]

**Review by bdauria — DISMISSED** [review]

**Review by dmyios — APPROVED** [review]

**Review by dmyios — APPROVED** [review]

**Review by bdauria — APPROVED** [review]

File: centreon/lang/de_DE.UTF-8/LC_MESSAGES/messages.po [hunk 1]
```diff
@@ -15810,3 +15810,30 @@ msgstr ""
...
+# msgid "Too many metrics, add more filter on resources"
+# msgstr ""
+
+# msgid "Service name"
+# msgstr ""
+
+# msgid "Please select a metric"
+# msgstr ""
+
+# msgid "The limit of 2 units has been reached"
```
Comment (smau-centreon) [inline centreon/lang/de_DE.UTF-8/LC_MESSAGES/messages.po:15838]:
```suggestion
# msgid "You have reached the limit of 2 units"
```

File: centreon/lang/es_ES.UTF-8/LC_MESSAGES/messages.po [hunk 2]
```diff
@@ -16145,3 +16145,30 @@ msgstr ""
 
 #  msgid "Host template required in service group association is not linked to service template"
```
Comment (smau-centreon) [inline centreon/lang/es_ES.UTF-8/LC_MESSAGES/messages.po:16146]:
```suggestion
#  msgid "Host template required in service group association is not linked to the service template"
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 3]
```diff
@@ -17967,6 +17967,33 @@ msgstr "Erreur lors de l'affichage des ressources des notifications"
 msgid "Host template required in service group association is not linked to service template"
```
Comment (smau-centreon) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:17967]:
```suggestion
msgid "Host template required in service group association is not linked to the service template"
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 4]
```diff
@@ -17967,6 +17967,33 @@ msgstr "Erreur lors de l'affichage des ressources des notifications"
...
+msgid "Too many metrics, add more filter on resources"
+msgstr "Trop de métriques, ajoutez plus de filtres sur les ressources"
+
+msgid "Service name"
+msgstr "Nom du service"
+
+msgid "Please select a metric"
+msgstr "Veuillez sélectionner une métrique"
+
+msgid "The limit of 2 units has been reached"
```
Comment (smau-centreon) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:17994]:
```suggestion
msgid "You have reached the limit of 2 units"
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 5]
```diff
@@ -17967,6 +17967,33 @@ msgstr "Erreur lors de l'affichage des ressources des notifications"
...
+msgstr "La limite de 2 unités a été atteinte"
```
Comment (smau-centreon) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:17995]:
```suggestion
msgstr "Vous avez atteint la limite de 2 unités"
```

File: centreon/lang/pt_BR.UTF-8/LC_MESSAGES/messages.po [hunk 6]
```diff
@@ -16609,3 +16609,30 @@ msgstr ""
 
 #  msgid "Host template required in service group association is not linked to service template"
```
Comment (smau-centreon) [inline centreon/lang/pt_BR.UTF-8/LC_MESSAGES/messages.po:16610]:
```suggestion
#  msgid "Host template required in service group association is not linked to the service template"
```

File: centreon/lang/pt_BR.UTF-8/LC_MESSAGES/messages.po [hunk 7]
```diff
@@ -16609,3 +16609,30 @@ msgstr ""
...
+# msgid "Too many metrics, add more filter on resources"
+# msgstr ""
+
+# msgid "Service name"
+# msgstr ""
+
+# msgid "Please select a metric"
+# msgstr ""
+
+# msgid "The limit of 2 units has been reached"
```
Comment (smau-centreon) [inline centreon/lang/pt_BR.UTF-8/LC_MESSAGES/messages.po:16637]:
```suggestion
# msgid "You have reached the limit of 2 units"
```

File: centreon/lang/pt_PT.UTF-8/LC_MESSAGES/messages.po [hunk 8]
```diff
@@ -16598,3 +16598,30 @@ msgstr "Você não salvou as alterações, deseja continuar?"
 
 #  msgid "Host template required in service group association is not linked to service template"
```
Comment (smau-centreon) [inline centreon/lang/pt_PT.UTF-8/LC_MESSAGES/messages.po:16599]:
```suggestion
#  msgid "Host template required in service group association is not linked to the service template"
```

File: centreon/lang/pt_PT.UTF-8/LC_MESSAGES/messages.po [hunk 9]
```diff
@@ -16598,3 +16598,30 @@ msgstr "Você não salvou as alterações, deseja continuar?"
...
+# msgid "Too many metrics, add more filter on resources"
+# msgstr ""
+
+# msgid "Service name"
+# msgstr ""
+
+# msgid "Please select a metric"
+# msgstr ""
+
+# msgid "The limit of 2 units has been reached"
```
Comment (smau-centreon) [inline centreon/lang/pt_PT.UTF-8/LC_MESSAGES/messages.po:16626]:
```suggestion
# msgid "You have reached the limit of 2 units"
```

**Review by smau-centreon — CHANGES_REQUESTED** [review]

File: centreon/lang/pt_BR.UTF-8/LC_MESSAGES/messages.po (hunk 7 above)
Comment (Thebarda) [reply centreon/lang/pt_BR.UTF-8/LC_MESSAGES/messages.po:16637]: Same comment as the previous PR

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po (hunk 5 above)
Comment (Thebarda) [reply centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:17995]: Same comment as the previous PR

**Review by smau-centreon — APPROVED** [review]

**Pull request merged**